lib/ex_aws/auth.ex

```diff
@@ -30,10 +30,19 @@ defmodule ExAws.Auth do
   def presigned_url(http_method, url, service, datetime, config, expires, query_params \\ []) do
     service = service_name(service)
     headers = presigned_url_headers(url)
-    query = presigned_url_query(service, datetime, config, expires, query_params)
-    url = "#{url}?#{query}"
-    signature = signature(http_method, url, headers, nil, service, datetime, config)
-    "#{url}&X-Amz-Signature=#{signature}"
+
+    org_query_params = query_params |> Enum.map(fn({k, v}) -> {to_string(k), v} end)
+    amz_query_params = build_amz_query_params(service, datetime, config, expires)
+    {org_query, amz_query} = [org_query_params, amz_query_params]
+    |> Enum.map(&(canonical_query_params(&1)))
+    |> List.to_tuple
```

Comment:
Thank your for your simple code. I fixed it. d452679cd9e00987e684608ecb1c475e614fb06a